subsys/bluetooth/controller/ll_sw/ull_df.c

```diff
@@ -689,4 +649,287 @@ static struct lll_df_adv_cfg *df_adv_cfg_acquire(void)
 
 	return df_adv_cfg;
 }
+
+/*
+ * @brief Function sets content of cte_info field in periodic advertising chain.
+ *
+ * The function allocates new PDU (or chain of PDUs) for periodic advertising to
+ * fill it with information about CTE that is going to be transmitted with the PDU.
+ * If there is already allocated PDU, it will be updated to hold information about
+ * CTE.
+ *
+ * @param adv           Pointer to periodic advertising set.
+ * @param df_cfg        Pointer to direction finding configuration
+ * @param[out] ter_idx  Pointer used to return index of allocated or updated PDU.
+ *                      Index is required for scheduling the PDU for transmission in LLL.
+ *
+ * @return Zero in case of success, other value in case of failure.
+ */
+static uint8_t set_cte_info_in_per_adv_chain(struct ll_adv_set *adv, struct lll_df_adv_cfg *df_cfg,
+					     uint8_t *ter_idx)
+{
+	struct adv_pdu_field_data pdu_data;
+	struct pdu_adv *pdu_prev, *pdu;
+	struct lll_adv_sync *lll_sync;
+	struct pdu_cte_info cte_info;
+	uint8_t pdu_add_field_flags;
+	void *extra_data;
+	uint8_t err;
+
+	lll_sync = adv->lll.sync;
+
+	cte_info.type = df_cfg->cte_type;
+	cte_info.time = df_cfg->cte_length;
+	pdu_data.field_data = (uint8_t *)&cte_info;
+	pdu_data.extra_data = df_cfg;
+
+	/* Note: ULL_ADV_PDU_HDR_FIELD_CTE_INFO is just information that extra_data
+	 * is required in case of this ull_adv_sync_pdu_alloc.
+	 * Other flags here do not change anything. It may be changed to use
+	 * true/false flag for extra_data allocation.
+	 */
+	err = ull_adv_sync_pdu_alloc(adv, ULL_ADV_PDU_HDR_FIELD_CTE_INFO, 0, NULL, &pdu_prev, &pdu,
+				     NULL, &extra_data, ter_idx);
+	if (err != 0) {
+		return err;
+	}
```

Comment:
Since `ull_adv_sync_pdu_alloc` returns an `BT_HCI_ERR_` code, we could change this check to 
`if (err != BT_HCI_ERR_SUCCESS)`, but that seems to not be done a lot in the controller (or at all), so perhaps it's best to be consistent.

Reply:
I've changed it to `if (err != BT_HCI_ERR_SUCCESS)`. Lets stay consistent about return types and macros used for that.